src/solver/optimisation/adequacy_patch_csr/hourly_csr_problem.h

```diff
@@ -34,9 +34,68 @@
 #include <antares/study/parameters/adq-patch-params.h>
 #include "../opt_structure_probleme_a_resoudre.h"
 
+
+
+struct LinkVariable
+{
+    LinkVariable() : directVar(-1), indirectVar(-1)
+    {
+    }
+    LinkVariable(int direct, int indirect) : directVar(direct), indirectVar(indirect)
+    {
+    }
+    inline bool check() const
+    {
+        if (directVar < 0)
+            Antares::logs.warning() << "directVar < 0 detected, this should not happen";
+        if (indirectVar < 0)
+            Antares::logs.warning() << "indirectVar < 0 detected, this should not happen";
+
+        return (directVar >= 0) && (indirectVar >= 0);
+    }
+    int directVar;
+    int indirectVar;
+};
+
 struct PROBLEME_HEBDO;
 class HourlyCSRProblem
```

Comment:
here we mainly : 
- make more data members of class **HourlyCSRProblem** private
-  move **struct LinkVariable** out from class **HourlyCSRProblem**